index.js

```diff
@@ -466,100 +565,89 @@ class GitlabScm extends Scm {
-    _decorateUrl({ scmUri, token }) {
-        return this.lookupScmUri({
+    async _decorateUrl({ scmUri, token }) {
+        const { hostname, owner, reponame, branch, rootDir } = await this.lookupScmUri({
             scmUri,
             token
-        }).then(({ hostname, owner, reponame, branch, rootDir }) => {
-            const baseUrl = `${hostname}/${owner}/${reponame}/-/tree/${branch}`;
-
-            return {
-                branch,
-                name: `${owner}/${reponame}`,
-                url: `${this.config.gitlabProtocol}://` +
-                    `${rootDir ? Path.join(baseUrl, rootDir) : baseUrl}`,
-                rootDir: rootDir || ''
-            };
         });
+        const baseUrl = `${hostname}/${owner}/${reponame}/-/tree/${branch}`;
+
+        return {
+            branch,
+            name: `${owner}/${reponame}`,
+            url: `${this.config.gitlabProtocol}://` +
+                `${rootDir ? Path.join(baseUrl, rootDir) : baseUrl}`,
+            rootDir: rootDir || ''
+        };
     }
 
     /**
      * Decorate the commit based on the repository
-     * @method _decorateCommit
+     * @async _decorateCommit
      * @param  {Object}        config            Configuration object
      * @param  {Object}        config.scmUri     SCM URI the commit belongs to
      * @param  {Object}        config.sha        SHA to decorate data with
      * @param  {Object}        config.token      Service token to authenticate with Github
      * @param  {Object}        config.scmContext Context to which user belongs
      * @return {Promise}
      */
-    _decorateCommit({ scmUri, sha, token, scmContext }) {
-        const commitLookup = this.lookupScmUri({
+    async _decorateCommit({ scmUri, sha, token }) {
+        const { owner, reponame } = await this.lookupScmUri({
             scmUri,
             token
-        }).then(scmInfo =>
-            this.breaker.runCommand({
-                json: true,
-                method: 'GET',
-                auth: {
-                    bearer: token
-                },
-                url: `${this.config.gitlabProtocol}://${this.config.gitlabHost}/api/v4` +
-                     `/projects/${scmInfo.owner}%2F${scmInfo.reponame}` +
-                     `/repository/commits/${sha}`
-            }).then((response) => {
-                checkResponseError(response, '_decorateCommit: commitLookup');
-
-                return {
-                    commitInfo: response.body,
-                    scmInfo
-                };
-            })
-        );
-
-        const authorLookup = commitLookup.then((commitData) => {
-            if (!commitData.commitInfo.author_name) {
-                return DEFAULT_AUTHOR;
-            }
+        });
 
-            return this.decorateAuthor({
-                token,
-                username: commitData.commitInfo.author_name,
-                scmContext
-            });
+        // There's no username provided, so skipping decorateAuthor
```

Comment:
how's it different from previous flow ? `commitData.commitInfo.author_name` no longer applicable ?

Reply:
don't think this was ever working in Gitlab, or they must have changed their return data
payload only returns these fields in terms of commiter/author data:
```
    "author_name": "Tiffany K",
    "author_email": "tiffanykyi@gmail.com",
    "authored_date": "2021-04-14T20:24:18.000+00:00",
    "committer_name": "Tiffany K",
    "committer_email": "tiffanykyi@gmail.com",
    "committed_date": "2021-04-14T20:24:18.000+00:00",
```
so `username` is never returned, so we cannot lookup additional author data since we don't have `username`